src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  shape:Shape;
+  color:Color;
 
+  getArea():number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color:Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    const isSideCorrect = (...args:number[]):boolean => {
+      return args.some((arg:number) => {
+        return arg <= 0;
+      });
+    };
+
+    if (isSideCorrect()) {
+      throw new Error('Incorrect side length parameter');
+    }
+
+    if ((a >= b + c) || (b >= a + c) || (c >= a + b)) {
+      throw new Error(`${a}, ${b} and ${c} can't be a triangle`);
+    }
+  }
 
+  getArea():number {
+    const getHalf: number = (this.a + this.b + this.c) / 2;
```

Comment:
You store in this variable `semiPerimetr` when I read `getHalf` I thought thet there should store a function that could help me to get a half. Because from verb starts names of the function. But here it isn't.